tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/ArgoCdClient.java

```diff
@@ -20,241 +20,180 @@
+                        setSocketTimeout((int) waitTimeout.toMillis()).build())
+                .build();
+        CloseableHttpResponse response = client.getHttpClient().execute(request);
+
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+        String line;
+        try {
+            while ((line = bufferedReader.readLine()) != null) {
+                StreamResultOfV1alpha1ApplicationWatchEvent result = objectMapper.readValue(line, StreamResultOfV1alpha1ApplicationWatchEvent.class);
+                if (result.getError() != null) {
+                    throw new RuntimeException("Error waiting for status: " + result.getError());
                 }
 
-                if (result.application() == null) {
+                if (result.getResult() == null) {
                     throw new RuntimeException("Error waiting for status: no application");
                 }
 
                 boolean operationInProgress = false;
 
-                Application a = result.application();
+                V1alpha1Application a = result.getResult().getApplication();
                 // consider the operation is in progress
-                if (a.operation() != null) {
+                if (a.getOperation() != null) {
                     // if it just got requested
                     operationInProgress = true;
-                    if (!Objects.requireNonNull(a.operation()).dryRun()) {
+                    if (Boolean.FALSE.equals(a.getOperation().getSync().getDryRun())) {
                         refresh = true;
                     }
-                } else if (a.status().operationState() != null) {
-                    Application.OperationState opState = Objects.requireNonNull(a.status().operationState());
-                    OffsetDateTime finishedAt = opState.finishedAt();
-                    OffsetDateTime reconciledAt = a.status().reconciledAt();
-                    Application.Operation operation = opState.operation();
+                } else if (a.getStatus().getOperationState() != null) {
+                    V1alpha1OperationState opState = Objects.requireNonNull(a.getStatus().getOperationState());
+                    OffsetDateTime finishedAt = OffsetDateTime.parse(opState.getFinishedAt());
+                    OffsetDateTime reconciledAt = OffsetDateTime.parse(a.getStatus().getReconciledAt());
+                    V1alpha1Operation operation = opState.getOperation();
 
                     if (finishedAt == null) {
                         // if it is not finished yet
                         operationInProgress = true;
-                    } else if (operation != null && !operation.dryRun() && (reconciledAt == null || reconciledAt.isBefore(finishedAt))) {
+                    } else if (operation != null && !Boolean.TRUE.equals(operation.getSync().getDryRun()) && (reconciledAt == null || reconciledAt.isBefore(finishedAt))) {
```

Comment:
`!Boolean.TRUE.equals(operation.getSync().getDryRun())` can be simply `!(operation.getSync().getDryRun())` ?

Reply:
This is to avoid null pointer exception when getDryRun() returns null. Previously it was https://github.com/walmartlabs/concord-plugins/pull/118/files#diff-b8ce43bb90950b31f5673aaa01cc5c1e6b8613c88792da533fe24fe49f07ad21L73